crates/optimism/chainspec/src/chain_metadata.rs

```diff
@@ -0,0 +1,312 @@
+use alloy_chains::NamedChain;
+use alloy_genesis::ChainConfig;
+use alloy_primitives::{ChainId, U256};
+use serde::{Deserialize, Serialize};
+
+/// The chain metadata stored in a superchain toml config file.
+/// Referring here as `ChainMetadata` to avoid confusion with `ChainConfig`.
+/// Find configs here: `<https://github.com/ethereum-optimism/superchain-registry/tree/main/superchain/configs>`
+/// This struct is stripped down to only include the necessary fields. We use JSON instead of
+/// TOML to make it easier to work in a no-std environment.
+#[derive(Clone, Debug, Deserialize)]
+#[serde(rename_all = "snake_case")]
+pub(crate) struct ChainMetadata {
+    pub chain_id: ChainId,
+    pub hardforks: HardforkConfig,
+    pub optimism: Option<OptimismConfig>,
+}
+
+#[derive(Clone, Debug, Deserialize)]
```

Comment:
this type already lives in kona, how do we want to dedup? @refcell @clabby @mattsse 
https://github.com/op-rs/kona/blob/a1899da1cfc0932b2f42fb799abdc5b558c1ba39/crates/protocol/genesis/src/chain/hardfork.rs#L3-L58

Reply:
yeah wen't need these types really, and would get away with minal bindings so that we get access to what we need to put into the genesis object